students/alex_whitty/lesson3/mailroom.py

```diff
@@ -0,0 +1,106 @@
+import sys  # imports go at the top of the file
+
+
+# Data Structure
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Elon Musk": [2263.23, 3300.87, 15432.0],
+            }
+
+
+
+# Input/Output
+prompt = "\n".join(("Thanking your donors for their generous gifts!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Exit",
+          ">>> "))
+
+# Features
+def thank_you():
+
+    answer = input("Please enter Full Name.>>>")
+    if answer.lower() == 'list':
+        for donor in donor_db.keys():
+            print(donor)
+    else:
+        if answer in donor_db:
+            current_donor(answer)
+        else:
+            add_donor()
+
+
+def current_donor(donor):
+
+    answer = input("Please enter donor Full Name.>>>")
+    if answer.title() in donor_db:
+        amount = input("Please enter donor amount.>>>")
+        donor_db[donor].append(float(amount))
+
+
+
+def sort_key(donor_db):
+    return donor_db[:]
+sorted(donor_db, key=sort_key)
+
+
+def add_donor():
+
+
+    new_donor = input("Name of the new donor to add?").title()
+    donor_amount = input("What is the donation amount?")
+    donor_db[new_donor] = [float(donor_amount)]
+    print(donor_db)
+    print('Thanks, {} for your generous donation.'.format(new_donor))
+
+
+
+def report():
+
+
+    rh = '{:20}{:10}{:10}{}'.format('Donor Name', '| Total Given |', ' Num Gifts |', ' Average Gift')
+    print(rh)
+    print('_____________________________________________________________')
+
+
+    r1 = '{:20}{:>12}{:>10}{:>18}'.format('William Gates, III', '$653784.49', '2', '$326892.24')
+    r2 = '{:20}{:>12}{:>10}{:>18}'.format('Mark Zuckerberg', '$16396.10', '3', '$5465.37')
+    r3 = '{:20}{:>12}{:>10}{:>18}'.format('Jeff Bezos', '$877.33', '1', '$877.33')
+    r4 = '{:20}{:>12}{:>10}{:>18}'.format('Paul Allen', '$708.42', '3', '$236.44')
+    r5 = '{:20}{:>12}{:>10}{:>18}'.format('Elon Musk', '$200.32', '1', '$200.32')
+
+    print(r1)
+    print(r2)
+    print(r3)
+    print(r4)
+    print(r5)
+
```

Comment:
I missed _report()_ the first time around. Right now, it's hardcoded with a list of donors. This will not work, since you need to be able to add new donors and/or donations for existing donor, so the report needs to be dynamically created from _donor_db_.